### DISCUSSION TITLE
! LaTeX Error: File `sjtuthesis.cls' not found

### DISCUSSION THREAD

**Discussion by jiangxiaoqiang** [sjtug/SJTUThesis#980] [提问]
**问题描述**:

when I compile the main.tex using xelatex, shows error : ! LaTeX Error: File `sjtuthesis.cls' not found

**系统信息**:

- OS版本(带版本号): Linux apline 3.18
- TeX 发行版(带版本号): TeX Live 2023

**日志信息**:

```
This is XeTeX, Version 3.141592653-2.6-0.999995 (TeX Live 2023) (preloaded format=xelatex 2023.5.24) 6 OCT 2023 07:50
entering extended mode
restricted \write18 enabled.
%&-line parsing enabled.
**/opt/data/project/2023/9/3d7e6e8948a949cdb99ebafebd717377/main.tex
(/opt/data/project/2023/9/3d7e6e8948a949cdb99ebafebd717377/main.tex
LaTeX2e <2022-11-01> patch level 1
L3 programming layer <2023-05-15>

! LaTeX Error: File `sjtuthesis.cls' not found.

Type X to quit or to proceed,
or enter new name. (Default extension: cls)

Enter file name:
! Emergency stop.


l.13 ^^M

End of file on the terminal!


Here is how much of TeX's memory you used:
23 strings out of 476760
559 string characters out of 5807904
1843482 words of memory out of 5000000
20291 multiletter control sequences out of 15000+600000
512287 words of font info for 32 fonts, out of 8000000 for 9000
1348 hyphenation exceptions out of 8191
19i,0n,29p,138b,17s stack positions out of 10000i,1000n,20000p,200000b,200000s
No pages of output.


```

the compile command may look like: `xelatex -synctex=1 main.tex `

**Comment by LogCreative** [comment]
In order to compile the thesis correctly, use `latexmk -xelatex main` instead of `xelatex main` directly. Or use `make all` or `Compile.bat`, see [README](https://github.com/sjtug/SJTUThesis/blob/master/README_en-US.md) file for more details.

FYI: #858

**Reply by jiangxiaoqiang** [reply]
is it possible to support the xelatex command compile directly?  I am strongly suggest that it should support this directly compile style by default. or need to change to compile by xelatex command by myself?

**Reply by LogCreative** [reply]
To complete the compilation, `xelatex` is not the full story. `latexmk` will help you run the command sequence automatically: `xelatex`, `biber`, `xelatex`, `xelatex`, which is good enough for most users.

If you want to make `xelatex` detect the template files, try to install the package globally in your TeX distribution (recommended) or moving all the template files under `texmf` directory to the root directory of this repo (not recommended, which will make the root directory messy).

For example, if you are using TeX Live as your TeX distribution, run the command to install globally:
```bash
cp -r texmf/* "$(kpsewhich --var-value TEXMFHOME)"
```

**Reply by majestichou** [reply]
I use MiKTex and there is no option of latexmk. What can i do？

**Reply by majestichou** [reply]
I usually use latexpdf option

**Reply by LogCreative** [reply]
Install `latexmk` through MiKTeX Console:

![image](https://github.com/sjtug/SJTUThesis/assets/61653082/68b6aff9-e072-401b-b73d-724a5281d836)

**Reply by LogCreative** [reply]
Or install the package globally. Copy the subfolder `texmf/tex` directory to the location:

![image](https://github.com/sjtug/SJTUThesis/assets/61653082/369342a7-b8cb-4e30-bc93-9982ccbb0634)

Then refresh the database:
![image](https://github.com/sjtug/SJTUThesis/assets/61653082/6a29bd62-7d06-4af2-90e7-86004fc7c3d0)

**Reply by majestichou** [reply]
I have installed latexmk through MiKTeX Console alreadly and make sure the subfolder texmf/tex directory to the MikTex install location.
It still does not work:
```
This is pdfTeX, Version 3.141592653-2.6-1.40.25 (MiKTeX 23.10) (preloaded format=pdflatex.fmt)
 restricted \write18 enabled.
entering extended mode
(main.tex
LaTeX2e <2023-06-01> patch level 1
L3 programming layer <2023-10-10>

! LaTeX Error: File `sjtuthesis.cls' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: cls)

Enter file name
```

**Reply by majestichou** [reply]
Finally, I solved the problem by copy subfolder "sjtutex" into the MikTex install location (D:\Miktex). The final sjtutex dir is in "D:\Miktex\tex\latex\sjtutex"

**Comment by sjtugzx** [comment]
I came across the same error when I compiled my black project on latex.sjtu.edu.cn. My compiler was selected as xeLatex.

**Reply by LogCreative** [reply]
Make sure you upload the file `.latexmkrc` as well, which tells `latexmk` the project-specific settings.

**Reply by sjtugzx** [reply]
I have added this file to my project, but still met the same mistake.
<img width="813" alt="截屏2024-03-26 10 35 09" src="https://github.com/sjtug/SJTUThesis/assets/20066885/15ff1315-3dc9-4fb2-8f15-82daf2e76bcd">
<img width="305" alt="截屏2024-03-26 10 34 57" src="https://github.com/sjtug/SJTUThesis/assets/20066885/9819f9df-522a-4cda-9fe6-b89f0a9d5fad">

<img width="272" alt="截屏2024-03-26 10 34 45" src="https://github.com/sjtug/SJTUThesis/assets/20066885/094475c2-7661-4686-b1ff-f4369c9ae02d">

**Reply by LogCreative** [reply]
`SJTUThesis-master` subdirectory is not necessary (even not correct for SJTUThesis), make sure everything (including `.latexmkrc`) is in the root directory.

**Reply by paipai-lu** [reply]
macOS TeXLive 在本地make all 编译未经改动的模板也出现这个问题

**Reply by paipai-lu** [reply]
已解决：才发现SJTUTeX已经加入CTAN，但我本地TeX Live没有更新过；现在的模板texmf里没东西
